src/gromacs/nblib/gmxsetup.h

```diff
@@ -71,13 +71,27 @@ struct NbvSetupUtil
 {
     NbvSetupUtil(SimulationState system, const NBKernelOptions& options);
 
+    //! Sets up and returns a GmxForceCalculator
+    std::unique_ptr<GmxForceCalculator> setupGmxForceCalculator();
+
+private:
+    //! Reads Topology data from the user-facing layer to translate to GMX data structures
     void unpackTopologyToGmx();
 
-    std::unique_ptr<nonbonded_verlet_t> setupNbnxmInstance();
+    //! Puts particles on a grid based on bounds specified by the box
+    void setParticlesOnGrid(std::unique_ptr<nonbonded_verlet_t>& nbv);
 
-    std::unique_ptr<GmxForceCalculator> setupGmxForceCalculator();
+    //! Sets Particle Types and Charges and VdW params
+    void setAtomProperties(std::unique_ptr<nonbonded_verlet_t>& nbv);
 
+    //! Sets up non-bonded verlet on the GmxFOrceCalculator
+    std::unique_ptr<nonbonded_verlet_t> setupNbnxmInstance();
+
+public:
+    //! Pointer to the SimulationState that contains topology, coordinates, velocities, particle/molecule data
```

Comment:
We do not need to document what a SimulationState is here.

Reply:
Done.